components/sup/src/manager/service/mod.rs

```diff
@@ -287,6 +289,9 @@ impl Service {
     }
 
     pub fn tick(&mut self, census_ring: &CensusRing) -> bool {
+        if self.desired_state == DesiredState::Down {
```

Comment:
This shouldn't be a necessary. Services which are in the `Down` state shouldn't be loaded so their tick function shouldn't be called